# 📊 Parcial – Análisis de Redes Sociales (Primer Corte)

## 🎯 Objetivo
Evaluar tu dominio de **filtros** y **agrupaciones** en Python (sin pivots ni gráficos) y tu capacidad de **análisis de negocio** en métricas de marketing: **CAC, LTV, churn rate, funnel, unit economics (LTV/CAC)**.

---
## 🧩 Contexto de Negocio (Caso Real)
**StartUp SaaS en crecimiento.** Vende planes **Basic, Pro, Enterprise** en **LatAm, North America, Europe, APAC**. Los canales de adquisición son: **meta_ads, google_ads, organic_search, outbound_sales**. Los usuarios se registran por **web** o **mobile_app**.

La dirección quiere decidir **dónde invertir el presupuesto del próximo trimestre**. Te piden:

1) Identificar **canales más rentables** (relación **LTV/CAC** y churn).  
2) Detectar **regiones** con mayor **potencial de crecimiento** y/o **riesgo**.  
3) Evaluar el desempeño por **tier** (Basic, Pro, Enterprise).  
4) Recomendar **asignación de presupuesto** por **canal + tier**.

---
## 📂 Instrucciones
- Trabaja únicamente con **filtros** y **agrupaciones**. **No uses** tablas dinámicas (pivot), merges ni gráficos.
- Archivo a usar: `clientes_marketing.csv`.
- Escribe **código + interpretación breve** para cada inciso.
- Si el enunciado dice **“Agrupa”**, usa `groupby`. Si dice **“Filtra”**, usa filtrado con máscaras booleanas. Si dice **“Ambas”**, realiza primero el **filtro** y luego la **agrupación**.

---


## Descargar Tabla (Correr una vez)

In [1]:
!wget https://github.com/javierherrera1996/IntroMarketingAnalytics/raw/refs/heads/main/PrimerCorte/cac_ltv_model.csv

--2025-09-17 23:40:40--  https://github.com/javierherrera1996/IntroMarketingAnalytics/raw/refs/heads/main/PrimerCorte/cac_ltv_model.csv
Resolving github.com (github.com)... 20.27.177.113
Connecting to github.com (github.com)|20.27.177.113|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://raw.githubusercontent.com/javierherrera1996/IntroMarketingAnalytics/refs/heads/main/PrimerCorte/cac_ltv_model.csv [following]
--2025-09-17 23:40:41--  https://raw.githubusercontent.com/javierherrera1996/IntroMarketingAnalytics/refs/heads/main/PrimerCorte/cac_ltv_model.csv
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.109.133, 185.199.110.133, 185.199.111.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.109.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 661398 (646K) [text/plain]
Saving to: ‘cac_ltv_model.csv’

cac_ltv_model.csv   100%[===================>] 645.90K  2.66MB

## Importar Datos

In [2]:
import pandas as pd
pd.set_option('display.float_format', lambda x: f'{x:,.2f}')

# 1) Cargar dataset
data = pd.read_csv('cac_ltv_model.csv')


### ✅ Variables relevantes (recordatorio)
- `acquisition_channel`, `signup_source`, `region`, `customer_tier`
- `plan_price`, `discount_rate`, `arpu`, `gross_margin`, `churn_rate`
- `contract_length_months`, `marketing_spend`

### 🧮 Fórmulas de negocio (a usar en el examen)
- **CAC por canal**:  
    $$ CAC = \frac{\text{Total Marketing Spend}}{\text{Clientes adquiridos}} $$
  
- **LTV por observación** (aprox.):  
   $$ LTV = ARPU \times contract\_length\_months \times gross\_margin \times (1 - churn\_rate) $$

- **Unit economics**:  
  $$ LTV/CAC   $$(ideal \> 3 como regla general de SaaS)

---



### Parte Exploratoria: Cree una muestra de los primeros 5 elementos de esta tabla. ¿Cuantas observaciones tienes?

In [3]:
data.head(5)

year  month    date  customer_id acquisition_channel signup_source  \
0  2023      1  Jan-23         1001      outbound_sales           web   
1  2023      1  Jan-23         1002            meta_ads           web   
2  2023      1  Jan-23         1003      organic_search    mobile_app   
3  2023      1  Jan-23         1004      organic_search           web   
4  2023      1  Jan-23         1005      organic_search           web   

          region customer_tier  plan_price  discount_rate   arpu  \
0  North America         Basic       78.84           0.10  63.63   
1          LatAm           Pro      120.23           0.00 124.47   
2  North America    Enterprise      335.82           0.20 278.20   
3         Europe           Pro      193.89           0.10 190.65   
4           APAC    Enterprise      471.80           0.10 445.32   

   gross_margin  churn_rate  contract_length_months  marketing_spend  
0          0.76        0.02                      12           212.48  
1          0.84        0.05                       1           352.60  
2          0.81        0.02                      12            55.40  
3          0.80        0.08                       1            49.10  
4          0.80        0.08                       1            45.27

Es una base de datos que tiene muchas adquisiciones en diferentes partes del mundo, así mismo distintos clientes

## 🔎 Parte A – Filtros

hint: Crea una tabla con el filtro y luego agrupa la variable que te piden.

1. **Filtra** los clientes del canal **`meta_ads`**. Calcula el **ARPU promedio** de este subconjunto.  



In [6]:
meta_ads = data[data['acquisition_channel'] == 'meta_ads']
arpu_promedio = meta_ads['arpu'].mean()
print("arpu promedio (meta_ads):",arpu_promedio)

arpu promedio (meta_ads): 168.78858272162617


In [7]:
meta_ads.head(5)

year  month    date  customer_id acquisition_channel signup_source  \
1   2023      1  Jan-23         1002            meta_ads           web   
10  2023      1  Jan-23         1011            meta_ads           web   
15  2023      1  Jan-23         1016            meta_ads    mobile_app   
19  2023      1  Jan-23         1020            meta_ads    mobile_app   
23  2023      1  Jan-23         1024            meta_ads    mobile_app   

           region customer_tier  plan_price  discount_rate   arpu  \
1           LatAm           Pro      120.23           0.00 124.47   
10  North America    Enterprise      341.61           0.00 342.49   
15         Europe    Enterprise      282.59           0.00 293.98   
19  North America           Pro      197.22           0.00 181.95   
23    Middle East           Pro      138.31           0.10 139.64   

    gross_margin  churn_rate  contract_length_months  marketing_spend  
1           0.84        0.05                       1           352.60  
10          0.89        0.07                       1           376.83  
15          0.85        0.07                       1           355.55  
19          0.76        0.03                      12           363.98  
23          0.82        0.08                       1           367.74

2. **Filtra** clientes de **LatAm** con **`churn_rate > 0.05`**. ¿Cuántos son? ¿Cuál es su **ARPU promedio**?  

In [10]:
churn_rate_latam = data[(data['region'] == 'LatAm') & (data['churn_rate'] > 0.05)]
arpu_promedio = meta_ads['arpu'].mean()
print("arpu promedio (meta_ads):",arpu_promedio)

arpu promedio (meta_ads): 168.78858272162617


3. **Filtra** clientes **Enterprise** con `contract_length_months > 6`. ¿Cuál es su **gross_margin promedio**?

In [11]:
contract_length_months = data[(data['customer_tier'] == 'Enterprise') & (data['contract_length_months'] > 6)]
gross_margin_promedio = contract_length_months['gross_margin'].mean()
gross_margin_promedio = meta_ads['gross_margin'].mean()
print("gross_margin promedio (meta_ads):",gross_margin_promedio)

gross_margin promedio (meta_ads): 0.8250197628458499


## 📊 Parte B – Agrupaciones (SOLO agrupar)
4. **Agrupa** por `acquisition_channel` y calcula **ARPU promedio**. Ordena de mayor a menor.  


In [19]:
acquisition_channel = data.groupby('acquisition_channel')['arpu'].mean().sort_values(ascending=False)
print(acquisition_channel)

acquisition_channel
outbound_sales   173.61
google_ads       172.37
organic_search   171.14
meta_ads         168.79
Name: arpu, dtype: float64


5. **Agrupa** por `region` y calcula **churn_rate promedio**. Identifica la región con mayor churn.  



In [20]:
region = data.groupby('region')['churn_rate'].mean().sort_values(ascending=False)
print(region)

region
Middle East     0.05
LatAm           0.05
North America   0.05
APAC            0.05
Europe          0.05
Africa          0.05
Name: churn_rate, dtype: float64


6. **Agrupa** por `customer_tier` y calcula **marketing_spend total**. ¿Cuál tier consume más presupuesto?

In [21]:
marketing_spend = data.groupby('customer_tier')['marketing_spend'].sum()
print(marketing_spend)

customer_tier
Basic        584,419.11
Enterprise   584,348.87
Pro          589,265.04
Name: marketing_spend, dtype: float64


## 🔀 Parte C – Filtro **y** Agrupación (Ambas)
7. **Filtra** solo registros de **`signup_source = 'web'`** y luego **agrupa** por `acquisition_channel` para obtener el **churn_rate promedio**.  


In [22]:
signup_source = data[data['signup_source'] == 'web']
churn_rate = signup_source.groupby('acquisition_channel')['churn_rate'].mean()
print(churn_rate)

acquisition_channel
google_ads       0.05
meta_ads         0.05
organic_search   0.05
outbound_sales   0.05
Name: churn_rate, dtype: float64


8. **Filtra** solo **`mobile_app`** y luego **agrupa** por `region` para calcular **ARPU promedio**.  


In [23]:
mobile_app = data[data['signup_source'] == 'mobile_app']
arpu = mobile_app.groupby('region')['arpu'].mean()
print(arpu)

region
APAC            169.09
Africa          172.60
Europe          165.89
LatAm           173.20
Middle East     171.71
North America   174.00
Name: arpu, dtype: float64


9. **Filtra** a clientes **`Pro`** y **agrupa** por `acquisition_channel` para calcular **marketing_spend total**.


In [26]:
customer_tier = data[data['customer_tier'] == 'Pro']
marketing_spend = customer_tirest.groupby('acquisition_channel')['marketing_spend'].sum()
print(marketing_spend)

acquisition_channel
google_ads       226,721.68
meta_ads         211,517.60
organic_search    29,848.72
outbound_sales   121,177.04
Name: marketing_spend, dtype: float64


## 📈 Parte D – Métricas de negocio (CAC, LTV, LTV/CAC)
10. **CAC por canal (Agrupa)**: calcula el CAC de cada `acquisition_channel` como:  
   `CAC = marketing_spend_total_del_canal / #clientes_del_canal`  


In [27]:
CAC= data.groupby('acquisition_channel')['marketing_spend'].sum() / data.groupby('acquisition_channel')['customer_id'].count()
print(CAC)

acquisition_channel
google_ads       401.02
meta_ads         349.27
organic_search    49.95
outbound_sales   200.22
dtype: float64


11. **CAC por tier (Agrupa)**: calcula el CAC de cada `customer_tier` como:  
   `CAC = marketing_spend_total_del_tier / #clientes_del_canal`  


In [28]:
CAC= data.groupby('customer_tier')['marketing_spend'].sum() / data.groupby('customer_tier')['customer_id'].count()
print(CAC)

customer_tier
Basic        249.75
Enterprise   249.19
Pro          248.43
dtype: float64


12. **LTV por tier (Ambas)**: crea una columna `ltv_individual` con la fórmula dada.


In [30]:
LTV = data['arpu'] * data['contract_length_months'] * data['gross_margin'] * (1 - data['churn_rate'])
data['ltv_individual'] = LTV
data.head(5)

year  month    date  customer_id acquisition_channel signup_source  \
0  2023      1  Jan-23         1001      outbound_sales           web   
1  2023      1  Jan-23         1002            meta_ads           web   
2  2023      1  Jan-23         1003      organic_search    mobile_app   
3  2023      1  Jan-23         1004      organic_search           web   
4  2023      1  Jan-23         1005      organic_search           web   

          region customer_tier  plan_price  discount_rate   arpu  \
0  North America         Basic       78.84           0.10  63.63   
1          LatAm           Pro      120.23           0.00 124.47   
2  North America    Enterprise      335.82           0.20 278.20   
3         Europe           Pro      193.89           0.10 190.65   
4           APAC    Enterprise      471.80           0.10 445.32   

   gross_margin  churn_rate  contract_length_months  marketing_spend  \
0          0.76        0.02                      12           212.48   
1          0.84        0.05                       1           352.60   
2          0.81        0.02                      12            55.40   
3          0.80        0.08                       1            49.10   
4          0.80        0.08                       1            45.27   

   ltv_individual  
0          567.54  
1           99.20  
2        2,651.91  
3          139.68  
4          328.72

12. A. Luego **agrupa** `ltv_individual` por
`acquisition_channel` para obtener el **LTV promedio**.  

In [31]:
LTV = data.groupby('acquisition_channel')['ltv_individual'].mean()
print(LTV)

acquisition_channel
google_ads       710.25
meta_ads         729.05
organic_search   765.13
outbound_sales   751.57
Name: ltv_individual, dtype: float64


12. B. Luego **agrupa** `ltv_individual`por `customer_tier` para obtener el **LTV promedio**.





In [32]:
LTV = data.groupby('customer_tier')['ltv_individual'].mean()
print(LTV)

customer_tier
Basic          291.77
Enterprise   1,349.02
Pro            577.63
Name: ltv_individual, dtype: float64


13. **Unit economics (Ambas)**: combina tus resultados para comparar **LTV promedio por tier** contra **CAC por canal** y comenta **qué combinaciones canal + tier** lucen más saludables (busca **LTV/CAC > 3**).


In [41]:
LTV = data.groupby(['acquisition_channel'])['ltv_individual'].mean()
LTV


acquisition_channel
google_ads       710.25
meta_ads         729.05
organic_search   765.13
outbound_sales   751.57
Name: ltv_individual, dtype: float64

In [39]:
CAC = data.groupby('acquisition_channel')['marketing_spend'].sum() / data.groupby('acquisition_channel')['customer_id'].count()
CAC

acquisition_channel
google_ads       401.02
meta_ads         349.27
organic_search    49.95
outbound_sales   200.22
dtype: float64

In [42]:
UNIT_ECONOMICS = LTV / CAC
print(UNIT_ECONOMICS)

acquisition_channel
google_ads        1.77
meta_ads          2.09
organic_search   15.32
outbound_sales    3.75
dtype: float64


In [43]:
LTV= data.groupby(['customer_tier'])['ltv_individual'].mean()
LTV

customer_tier
Basic          291.77
Enterprise   1,349.02
Pro            577.63
Name: ltv_individual, dtype: float64

In [44]:
CAC= data.groupby('customer_tier')['marketing_spend'].sum() / data.groupby('customer_tier')['customer_id'].count()
CAC

customer_tier
Basic        249.75
Enterprise   249.19
Pro          248.43
dtype: float64

In [45]:
UNIT_ECONOMICS = LTV / CAC
print(UNIT_ECONOMICS)

customer_tier
Basic        1.17
Enterprise   5.41
Pro          2.33
dtype: float64


## 🧠 Parte E – Análisis crítico (respuesta abierta)
13. **CMO por un día**: con tus resultados, ¿en qué **canal(es)** invertirías más el próximo trimestre? ¿Por qué? (Cita **CAC** y **churn**).  
14. ¿Qué **región** ves con mayor riesgo? ¿Qué hipótesis explicarían su **churn**?  
15. Identifica **dos combinaciones canal + tier** con mejor **LTV/CAC** y propón **dos acciones** concretas de optimización (p. ej., creatividades, audiencias, pricing, retención).

---
✍️ **Nota**: Mantente disciplinado con el enunciado (**Filtra**, **Agrupa**, **Ambas**). La evaluación pondera **correctitud técnica** y **calidad de interpretación de negocio**.


13) En la que más invertiría es Outbound_sales y Google_ads, ya que estos presentan los ARPU promedio más altos, además si se hace un contraste con CAC estos canales ofrecen una relación más eficas entre el costo de adquisición y el valor generado por el cliente. Otro factor a tener en cuenta es que el churn en este caso es menor al de los demás canales como meta_ads, lo que esto demuestra es una mayor retención.
14) La región con mayor resgo es LatAm, porque muestra que el churn-rate promedio es más elevado en comparación con las demás regiones. Las posibles hipótesis que se podrían explicar, los factores economicos la votalidad del ingreso disponible en clientes, lo cual aumenta la rotación. Por otro lado la competencia local donde existen alternativas más baratas que atraen más fácil a los usuarios.
15) La primera combinación: Outbound_sales + Enterprise
-LTVC/CAC=5.41, el mejor de todos los tires.
- Es decir que loc clientes con contratos más largos y mayor valor de vida.
Las dos acciones:
- Diseñar programas de fidelización exclusivos
- Incentivar contratos anuales con descuentos que sean de manera progresiva para así poder reducir el churn.
La segunda combinación: Google_ads + Pro
- LTVC/CAC=2.33, es atractivo y estable en canales digitales
- Tiene un buen alcance y balance entre costo y retorno
Acciones:
-Optimizar la segmentación de audiencia en Google_Ads para reducir el CAC
- Implementar productos complementarios para poder elevar la ARPU.